src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,71 @@
+package duke;
+
+import duke.commands.Command;
+import duke.tasktypes.TaskList;
+import java.util.Scanner;
+
+
+/**
+ * Represents the chat bot.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    /**
+     * Constructor for the class.
+     * @param filePath filePath for the data file.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        taskList = new TaskList(storage.load());
+    }
+
+
+    /**
+     * Starts the Dory chat bot.
+     */
+    public void run() {
+        ui.showIntro();
+
+        Scanner input = new Scanner(System.in);
+        while (input.hasNextLine()) {
+            try {
+                String beforeEdit = input.nextLine();
+                String nextInput = beforeEdit.toLowerCase();
+                ui.endOfCommand();
+                Command c = Parser.parse(nextInput);
+                c.execute(taskList, ui, storage);
+                storage.updateHardDisk(taskList);
+            } catch (Exception e) {
+                ui.showError(e);
+            } finally {
+                ui.endOfResponse();
+            }
+        }
+    }
+
+    /**
+     * Generates response for user's input.
+     * @param input
```

Comment:
I'm not sure, but maybe its better for javadocs to include a description for input too, to help ppl immediately grasp the kind of parameters you are looking for? (e.g. input user command typed in console) I'm looking [here](https://se-education.org/guides/conventions/java/intermediate.html#comments)

I noticed this in the other javadocs too, so remember to fix those too if this comment makes sense.

```suggestion
     * Generates response for user's input.
     *
     * @param input user command typed in console
```